docs/references/astro/locals.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Locals
+description: Learn how to authenticate your Astro application with Clerk using locals.
+---
+
+# Locals
+
+Through Astro [`locals`](https://docs.astro.build/en/guides/middleware/#storing-data-in-contextlocals), the [`Auth`](/docs/references/nextjs/auth-object) and current user objects can be accessed between middlewares and pages. These locals are injected when you install the provided [middleware](/docs/references/astro/clerk-middleware).
+
+The following guide provides examples of using the `auth()` local to validate an authenticated user and the `currentUser()` local to access the `Backend API User` object for the authenticated user.
+
+## Protect your pages
+
+You can use the `auth()` local to protect your pages. This local will return the current user's ID if they are signed in, or `null` if they are not.
+
+```astro filename="src/pages/protected.astro"
+---
+if (!Astro.locals.auth().userId) {
+  return Astro.redirect('/login');
+}
+---
+
+<div>Protected page</div>
+```
+
+## Protect your forms
+
+```astro filename="src/pages/form.astro"
+---
+if (Astro.request.method === "POST") {
+  if (!Astro.locals.auth().userId) {
+    throw new Error('You must be signed in to add an item to your cart');
+  }
+
+  const data = await Astro.request.formData();
+  console.log('add item action', data);
+}
+---
+
+<form method="POST">
+  <input
+    value="test"
+    type="text"
+    name="name"
+  />
+  <button type="submit">Add to Cart</button>
+</form>
+```
+
+## Accessing the current user
+
+Current user data is important for data enrichment. You can use the `currentUser()` local to fetch the current user's data in your pages.
+
```

Comment:
```suggestion

Under the hood, this local:
- calls `fetch()`, so it is automatically deduped per request.
- uses the `GET /v1/users/me` endpoint.
- counts towards the [Backend API Request Rate Limit](/docs/backend-requests/resources/rate-limits#rate-limits).
```

This might use something other than fetch. If so, please adjust.

Reply:
Pantelis addressed this